ckanext/canada/ati.py

```diff
@@ -55,56 +66,82 @@ def command(self):
+        :type csv_files: sequence of str
+
+        :return: Nothing
+        :rtype: None
+        """
+
+        conn = solr_connection('ati_summaries')
         lc = LocalCKAN()
-        for org in lc.action.organization_list():
-            count = 0
-            org_detail = lc.action.organization_show(id=org)
-            for records in _ati_summaries(org_detail, lc):
-                _update_records(records, org_detail, conn)
-                count += len(records)
-
-            print org, count
-        #rval = conn.query("*:*" , rows=2)
-
-def _solr_connection():
-    from solr import SolrConnection
-    url = config['ati_summaries.solr_url']
-    user = config.get('ati_summaries.solr_user')
-    password = config.get('ati_summaries.solr_password')
-    if user is not None and password is not None:
-        return SolrConnection(url, http_user=user, http_pass=password)
-    return SolrConnection(url)
-
-def _ati_summaries(org, lc):
-    """
-    generator of ati summary dicts for organization with name org
-    """
-    for dataset_type in DATASET_TYPES:
-        result = lc.action.package_search(
-            q="type:%s owner_org:%s" % (dataset_type, org['id']),
-            rows=1000)['results']
-        resource_id = result[0]['resources'][0]['id']
-        offset = 0
-        while True:
-            rval = lc.action.datastore_search(resource_id=resource_id,
-                limit=BATCH_SIZE, offset=offset)
-            records = rval['records']
-            if not records:
-                break
-            yield records
-            offset += len(records)
+        if csv_files:
+            count = {}
+            for csv_file in csv_files:
+                try:
+                    print csv_file + ':'
+                    count[csv_file] = {}
+                    for org_recs in csv_data_batch(csv_file, TARGET_DATASET):
+                        org_id = org_recs.keys()[0]
+                        if org_id not in count[csv_file]:
+                            count[csv_file][org_id] = 0
+                        org_detail = lc.action.organization_show(id=org_id)
+                        records = org_recs[org_id]
+                        _update_records(records, org_detail, conn)
+                        count[csv_file][org_id] += len(records)
+                    for k, v in count[csv_file].iteritems():
+                        print "    {0:s} {1}".format(k, v)
+                except _csvError as e:
+                    logging.error('On {0:s}, encountered: {1:s}'.format(
+                        csv_file,
+                        e.message))
+                except AssertionError as e:
+                    logging.warning('On {0:s}, encountered: {1:s}'.format(
+                        csv_file,
+                        e.message))
+                except IOError as e:
+                    logging.error('On {0:s}, encountered: {1:s}'.format(
+                        csv_file,
+                        e.strerror))
```

Comment:
This swallows the tracebacks and only keeps the error message. I'm ok with letting the traceback rise normally and cancel the whole process. We don't want to ignore errors here (and we certainly want all the information necessary to fix them).  Please remove the except: clauses here.